third_party/move/move-compiler-v2/src/pipeline/control_flow_graph_simplifier.rs

```diff
@@ -0,0 +1,687 @@
+            .remove(&block_to_remove)
+            .expect("successors");
+        self.code_blocks.remove(&block_to_remove);
+    }
+
+    /// Removes `pred` from the predecessors of `block`
+    fn remove_pred(&mut self, block: BlockId, pred: BlockId) {
+        let preds = self.predecessors.get_mut(&block).expect("predecessors");
+        preds.retain(|p| *p != pred);
+    }
+
+    /// Replaces the predecessor `old_pred` of `block` by `new_pred`
+    fn replace_preds(&mut self, block: BlockId, old_pred: BlockId, new_pred: BlockId) {
+        Self::replace_blocks(self.preds_mut(block), old_pred, new_pred);
+    }
+
+    /// Replaces the predecessor `old_pred` of `block` by `new_pred`
+    fn replace_succs(&mut self, block: BlockId, old_succ: BlockId, new_succ: BlockId) {
+        Self::replace_blocks(self.successors_mut(block), old_succ, new_succ);
+    }
+
+    /// Replaces `old` by `new` in `blocks`
+    fn replace_blocks(blocks: &mut [BlockId], old: BlockId, new: BlockId) {
+        for block in blocks {
+            if *block == old {
+                *block = new;
+            }
+        }
+    }
+
+    /// Checks if the control flow graph is consistent
+    /// - `self.code_blocks`, `self.successors`, and `self.predecessors` have the same set of keys
+    /// - `self.successors` is the reverse of `self.predecessors`
+    /// - TODO: check if the code blocks are consistent with the successors and predecessors map
+    fn check_consistency(&self) {
+        assert!(self.code_blocks.keys().collect_vec() == self.successors.keys().collect_vec());
+        assert!(self.code_blocks.keys().collect_vec() == self.predecessors.keys().collect_vec());
+        let pred_map_reversed = self.predecessors.iter().fold(
+            BTreeMap::new(),
+            |mut acc: BTreeMap<BlockId, Vec<BlockId>>, (block, preds)| {
+                acc.entry(*block).or_default();
+                for pred in preds {
+                    acc.entry(*pred).or_default().push(*block);
+                }
+                acc
+            },
+        );
+        assert!(self.successors.keys().collect_vec() == pred_map_reversed.keys().collect_vec());
+        for block in self.successors.keys() {
+            assert!(
+                self.successors
+                    .get(block)
+                    .expect("successors")
+                    .iter()
+                    .collect::<BTreeSet<_>>()
+                    == pred_map_reversed
+                        .get(block)
+                        .expect("pred_map_reversed")
+                        .iter()
+                        .collect::<BTreeSet<_>>()
+            );
+        }
+    }
+}
+
+/// Transformation state for transformation 1 in the module doc
+struct EmptyBlockRemover(ControlFlowGraphCodeGenerator);
+
+impl EmptyBlockRemover {
+    /// Performs the transformation 1 described in the module doc
+    pub fn transform(generator: ControlFlowGraphCodeGenerator) -> ControlFlowGraphCodeGenerator {
```

Comment:
```suggestion
        match block_instrs.len() {
            1 => matches!(block_instrs[0], Bytecode::Label(..))
            2 => matches!((block_instrs[0], block_instrs[1]), (Bytecode::Label(..), Bytecode::Jump(..))),
            _ => false
        }
```

Might be a bit easier to parse for the reader.

Reply:
done